android/KMEA/app/src/main/java/com/tavultesoft/kmea/cloud/CloudDownloadMgr.java

```diff
@@ -56,7 +57,13 @@ public synchronized void initialize(Context aContext)
   {
     if(isInitialized)
       return;
-    aContext.registerReceiver(completeListener,new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
+    try {
+      aContext.registerReceiver(completeListener, new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
+    } catch (Exception e) {
+      String message = "CloudDownloadMgr re-initializing";
+      Toast.makeText(aContext, message,
```

Comment:
A couple of things:

1. What is the specific exception? We should be trapping only that. 
2. I don't think it's helpful to show the user what is essentially a debug message. Logging it internally would be fine.

Reply:
got it - I'll update to catch `IllegalStateException`